botorch/generation/sampling.py

```diff
@@ -254,82 +252,101 @@ def __init__(
-                is a GP model for one constraint function,
-                or a MultiTaskGP model where each task is one
-                constraint function
-                All constraints are of the form c(x) <= 0.
-                In the case when the constraint model predicts
-                that all candidates violate constraints,
-                we pick the candidates with minimum violation.
-            minimize_constraints_only: False by default, if true,
-                we will automatically return the candidates
-                with minimum posterior constraint values,
-                (minimum predicted c(x) summed over all constraints)
-                reguardless of predicted objective values.
+            constraint_model: either a ModelListGP where each submodel is a GP model for
+                one constraint function, or a MultiTaskGP model where each task is one
+                constraint function. All constraints are of the form c(x) <= 0. In the
+                case when the constraint model predicts that all candidates
+                violate constraints, we pick the candidates with minimum violation.
         """
+        if objective is not None:
+            raise NotImplementedError(
+                "`objective` is not supported for `ConstrainedMaxPosteriorSampling`."
+            )
+
         super().__init__(
             model=model,
             objective=objective,
             posterior_transform=posterior_transform,
             replacement=replacement,
         )
         self.constraint_model = constraint_model
-        self.minimize_constraints_only = minimize_constraints_only
+
+    def _convert_samples_to_scores(self, Y_samples, C_samples) -> Tensor:
+        r"""Convert the objective and constraint samples into a score.
+
+        The logic is as follows:
+            - If a realization has at least on feasible candidate we use the objective
+                value as the score and set all infeasible candidates to -inf.
+            - If a realization doesn't have a feasible candidate we set the score to
+                the negative total violation of the constraints to incentivize choosing
+                the candidate with the smallest constraint violation.
+
+        Args:
+            Y_samples: A `num_samples x batch_shape x num_cand x 1`-dim Tensor of
+                samples from the objective function.
+            C_samples: A `num_samples x batch_shape x num_cand x num_constraints`-dim
+                Tensor of samples from the constraints.
+
+        Returns:
+            A `num_samples x batch_shape x num_cand x 1`-dim Tensor of scores.
+        """
+        is_feasible = (C_samples <= 0).all(
+            dim=-1
+        )  # num_samples x batch_shape x num_cand
+        has_feasible_candidate = is_feasible.any(dim=-1)
+
+        scores = Y_samples.clone()
+        scores[~is_feasible] = -float("inf")
+        if (~has_feasible_candidate).any():
```

Comment:
This reads more intuitively. ALso it's a micro optimization that saves a few bit flips :)
```suggestion
        if not has_feasible_candidate.all():
```

Reply:
while we're at the topic of micro-optimziation, you may want to just flip the bits and  assign `~has_feasible_candidate` to a variable since you're using it repeatedly here